conn_pool.go

```diff
@@ -119,23 +162,9 @@ func (p *ConnPool) acquire(deadline *time.Time) (*Conn, error) {
 		p.logger.Warn("All connections in pool are busy - waiting...")
 	}
 
-	// Set initial timeout/deadline value. If the method (acquire) happens to
-	// recursively call itself the deadline should retain its value.
-	if deadline == nil && p.acquireTimeout > 0 {
-		tmp := time.Now().Add(p.acquireTimeout)
-		deadline = &tmp
-	}
-	// If there is a deadline then start a timeout timer
-	if deadline != nil {
-		timer := time.AfterFunc(deadline.Sub(time.Now()), func() {
-			p.cond.Signal()
-		})
-		defer timer.Stop()
-	}
-
 	// Wait until there is an available connection OR room to create a new connection
 	for len(p.availableConnections) == 0 && len(p.allConnections) == p.maxConnections {
```

Comment:
since allConnections can be > maxConnections, should we change this 2nd condition to check:
```go
len(p.allConnections) >= p.maxConnections
```